components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/listener/TenantManagementListener.java

```diff
@@ -41,6 +41,7 @@ public void onTenantCreate(TenantInfoBean tenantInfoBean) throws StratosExceptio
         PrivilegedCarbonContext.getThreadLocalCarbonContext().startTenantFlow();
         PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantDomain(tenantDomain);
         PrivilegedCarbonContext.getThreadLocalCarbonContext().setTenantId(tenantInfoBean.getTenantId());
+        PrivilegedCarbonContext.endTenantFlow();
```

Comment:
we can completely remove this listener if there is no any usage, or else at least this method, as it do nothing

Reply:
We are setting the tenant domain and the tenant id to the thread local here it seems. I didn't get why we could remove the method.

Reply:
We cannot remove this listener as we delete recovery data during the tenant deletion: https://github.com/wso2-extensions/identity-governance/blob/b708db617d057c3e57d440b813cbc4f7c5298a42/components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/listener/TenantManagementListener.java#L91

also we cannot remove the onTenantCreate method since TenantManagementListener implements TenantMgtListener interface